### PR TITLE
Manual `pre-commit` update

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

[Black allows itself to tweak its style in the beginning of the year](https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy) and it tried to apply its updated style to `astropy` in #15995, but that pull request was not merged because we wanted to avoid letting Black make a non-negligible number of changes just before we switched from Black to Ruff formatter. Now Ruff formatter has caught up with this year's Black style and is making many of the same changes that Black tried to make in #15995. The `pre-commit` autoupdate in #16153 was closed to enable applying three additional manual formatting changes, which I have highlighted below.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks!!!

File: astropy/visualization/tests/test_time.py [hunk 1]
```diff
@@ -171,10 +171,11 @@ def test_formats(format, expected):
...
         # which ERFA emits a warning about the date being dubious
-        with pytest.warns(ErfaWarning) if format in [
-            "unix",
-            "plot_date",
-        ] else nullcontext():
+        with (
+            pytest.warns(ErfaWarning)
+            if format in ["unix", "plot_date"]
+            else nullcontext()
+        ):
```
Comment (eerovaher) [inline astropy/visualization/tests/test_time.py:178]: I removed a trailing comma, which would have forced Ruff to format this as
```patch
-        with pytest.warns(ErfaWarning) if format in [
-            "unix",
-            "plot_date",
-        ] else nullcontext():
+        with (
+            pytest.warns(ErfaWarning)
+            if format
+            in [
+                "unix",
+                "plot_date",
+            ]
+            else nullcontext()
+        ):
```

File: astropy/visualization/tests/test_time.py [hunk 2]
```diff
@@ -191,10 +192,11 @@ def test_auto_formats(format, expected):
...
```
Comment (eerovaher) [inline astropy/visualization/tests/test_time.py:199]: The trailing comma would have forced Ruff to format this as
```patch
-        with pytest.warns(ErfaWarning) if format in [
-            "unix",
-            "plot_date",
-        ] else nullcontext():
+        with (
+            pytest.warns(ErfaWarning)
+            if format
+            in [
+                "unix",
+                "plot_date",
+            ]
+            else nullcontext()
+        ):
```

**Review by nstarman — APPROVED** [review]
LGTM. Mostly white space changes and improved line breaking.

**Comment by eerovaher** [comment]
I have rebased as requested by https://groups.google.com/g/astropy-dev/c/2y2vVj1go0w

**Review by taldcroft — APPROVED** [review]
Fine for time, table, io.ascii.

**Comment by eerovaher** [comment]
The `pre-commit-ci` bot has already opened two fairly large pull requests that tried to make `astropy` comply with the Black 2024 style. If this pull request is not merged soon we will get a third such automatic pull request.

**Review by mhvk — APPROVED** [review]
I looked through everything up to units as well now, and this really is all fine. Let's get it in!

**Comment by mhvk** [comment]
p.s. Better than "fine" actually - the ruff version of black style deviates a tiny bit, but really in the right direction of not allowing strange line breaks.

**Comment by eerovaher** [comment]
As I've already mentioned, most (if not all) changes in this pull request are caused by Ruff formatter adopting Black 2024 style, not by differences between current Black and current Ruff formatter. See also [Black changelog](https://black.readthedocs.io/en/stable/change_log.html#id8) or see the changes Black tried to introduce in #15995.

**Comment by mhvk** [comment]
Even better then, that both black and ruff are getting rid of the few rough edges!  And thanks for doing this - you clearly had already had done a careful look at the changes.

**Pull request merged**